### PR TITLE
DOMMatrix: Tests for basic non-mutating methods

### PR DESCRIPTION
**Pull request by peterjoel**
This change is [Reviewable](https://reviewable.io/reviews/w3c/csswg-test/1066)

### CODE REVIEW

File: geometry-1/DOMMatrix-003.htm [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        function identity() {
+          return new DOMMatrix(
+            [1, 0, 0, 0,
+             0, 1, 0 ,0,
+             0, 0, 1, 0,
+             0, 0, 0, 1]);
+        }
+
+        function update(matrix, f) {
+          return f(matrix), matrix;
```
Comment (zcorpan) [inline geometry-1/DOMMatrix-003.htm]: Please avoid clever tricks like this in tests. :-)

**Comment by zcorpan** [comment]
Generally looks good an I'm happy to have this merged, but some thoughts for improvements (can be followup PR(s) if you like or file an issue for this):

* Factor out common code between tests to an external script file
* Test that the object you're calling on is not mutated (this is tested for DOMMatrixReadOnly in 002 but could be tested for DOMMatrix as well)
* Test all variants from 0 arguments to 1 too many arguments (excess arguments should be ignored per WebIDL)
* Test that is2D is set to false when it should and not when it shouldn't
* Check that the returned object is a DOMMatrix and not a DOMMatrixReadOnly
* Test both DOMMatrix and DOMMatrixReadOnly (these methods do the same thing for both so could run each test on both without duplicating test code)

**Comment by zcorpan** [comment]
One more:

* Test that a new object is returned for two subsequent calls to a method (with the same arguments). ([bug 29541](https://www.w3.org/Bugs/Public/show_bug.cgi?id=29541))

File: geometry-1/DOMMatrix-003.htm (hunk 1 above)
Comment (peterjoel) [inline geometry-1/DOMMatrix-003.htm]: You mean the update function itself, or returning with a comma?

**Comment by peterjoel** [comment]
Thanks for your comments @zcorpan. I'm going to follow up with more tests, which should cover your points as well as some methods that aren't covered there.

**Comment by peterjoel** [comment]
On the subject of factoring common code into an external script file, I didn't see any examples of that. Is there a particular pattern to follow, or is it just a matter of including a `.js` file in the same directory?

**Comment by zcorpan** [comment]
I think typically in a `support` directory.

File: geometry-1/DOMMatrix-003.htm (hunk 1 above)
Comment (peterjoel) [inline geometry-1/DOMMatrix-003.htm]: @zcorpan

Comment (zcorpan) [inline geometry-1/DOMMatrix-003.htm]: Returning with a comma

**Comment by zcorpan** [comment]
LGTM. Can you squash?

**Comment by peterjoel** [comment]
@zcorpan squashed

**Pull request merged**